File: code/client/src/app/entities/user/model/usuario.ts
export class Usuario {
  id?: string;
  carnet?: string | null;
  nombre?: string | null;
  apellido_materno?: string | null;
  apellido_paterno?: string | null;
  rol?: string | null;
  carrera_Id?: number | null;
  carrera?: string | null;
  correo?: string | null;
  telefono?: string | null;
  nombre_referencia?: string | null;
  telefono_referencia?: string | null;
  email_referencia?: string | null;
  bloqueado?: boolean;
  motivo_bloqueo?: string | null;
  imagen_frente_carnet?: string | null;
  imagen_atras_carnet?: string | null;
  constructor() {
    this.id = '';
    this.carnet = null;
    this.nombre = null;
    this.apellido_materno = null;
    this.apellido_paterno = null;
    this.rol = null;
    this.carrera_Id = null;
    this.carrera = null;
    this.correo = null;
    this.telefono = null;
    this.nombre_referencia = null;
    this.telefono_referencia = null;
    this.email_referencia = null;
    this.imagen_frente_carnet = null;
    this.imagen_atras_carnet = null;
  }
}

File: code/server/src/Core/Entities/Usuario.cs
using IMT_Reservas.Server.Core.Abstraction;

namespace IMT_Reservas.Server.Core.Entities;

public class Usuario : Entity
{
    public string Carnet { get; set; } = string.Empty;
    public string Nombre { get; set; } = string.Empty;
    public string ApellidoPaterno { get; set; } = string.Empty;
    public string ApellidoMaterno { get; set; } = string.Empty;
    public TipoUsuario Rol { get; set; } = TipoUsuario.Estudiante;
    public string Contrasena { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string Telefono { get; set; } = string.Empty;
    public string? TelefonoReferencia { get; set; }
    public string? NombreReferencia { get; set; }
    public string? EmailReferencia { get; set; }
    public int IdCarrera { get; set; }
    public bool Bloqueado { get; set; }
    public string? MotivoBloqueo { get; set; }
    public byte[]? ImagenFrenteCarnet { get; set; }
    public byte[]? ImagenAtrasCarnet { get; set; }
    public byte[]? ImagenFirma { get; set; }
    public string? RefreshToken { get; set; }
    public DateTime? RefreshTokenExpiry { get; set; }
    public bool EmailVerificado { get; set; } = true;
    public string? GoogleId { get; set; }
    public string? TokenVerificacionHash { get; set; }
    public DateTime? TokenVerificacionExpira { get; set; }
}
